### PR TITLE
http: Detect remote disconnect

### PR DESCRIPTION
**Pull request by promag**
Builds on #19420, review last commit only.

### CODE REVIEW

**Comment by promag** [comment]
This is a requirement in case we want to support RPC interruption due to connection close. For instance, currently interrupting the command `bitcoin-cli -regtest waitfornewblock` doesn't interrupt the server handling.

**Comment by promag** [comment]
Re-run CI.

File: test/functional/mining_getblocktemplate_longpoll.py [hunk 1]
```diff
@@ -22,7 +22,10 @@ def __init__(self, node):
         self.node = get_rpc_proxy(node.url, 1, timeout=600, coveragedir=node.coverage_dir)
 
     def run(self):
-        self.node.getblocktemplate({'longpollid': self.longpollid, 'rules': ['segwit']})
+        try:
+            self.node.getblocktemplate({'longpollid': self.longpollid, 'rules': ['segwit']})
+        except:
```
Comment (luke-jr) [inline test/functional/mining_getblocktemplate_longpoll.py:27]: Why is this test breaking?

**Comment by MarcoFalke** [comment]
Looks like this might be waiting on #19420

**Comment by promag** [comment]
@MarcoFalke 👍

**Comment by fanquake** [comment]
Given that #19420 (base PR) has now been closed, going to close this for now as well. If the base PR is reopened, we could reopen this PR too.

**Pull request closed without merging**